src/pages/LoginPage/Scanner/Login.tsx

```diff
@@ -40,7 +42,11 @@ const Login: FC = () => {
     event.preventDefault();
     if (!rfid.trim()) return;
     const user = await handleRfid(rfid);
-    if (user) dispatch(setUser(user));
+    if (!user) {
+      dispatch(setModalState(modalTypes.FAILURE));
+      return;
+    }
+    if (user.pk >= 0) dispatch(setUser(user));
```

Comment:
In which scenarios is line 49 intended to take effect?

Reply:
Check the other comment, but when handleRfid returns null (!user), there has been an error